third_party/move/move-core/types/src/value.rs

```diff
@@ -486,7 +493,22 @@ impl<'d> serde::de::DeserializeSeed<'d> for &MoveTypeLayout {
                 AccountAddress::deserialize(deserializer).map(MoveValue::Address)
             },
             MoveTypeLayout::Signer => {
-                AccountAddress::deserialize(deserializer).map(MoveValue::Signer)
+                let (_, fields) = MoveStructLayout::signer()
```

Comment:
Should we enforce a check that tag is 0 or 1? We seem to use these a lot in Value impl, I guess we can check at serde time to be more defensive and ensure if we add a variant, things do not break.